packages/worker/src/dependencies.ts

```diff
@@ -2,101 +2,47 @@
  * Type definitions for the Worker end of the external dependencies mechanism.
  *
  * External dependencies are functions injected into a Workflow isolate from the main Node.js isolate.
- * They are an advanced feature and should be used with care.
  *
- * @see proposal at https://github.com/temporalio/proposals/blob/master/node/logging-and-metrics-for-user-code.md
+ * Dependency functions may not return values to the Workflow in order to prevent breaking determinism.
```

Comment:
```suggestion
 * Dependency functions may not return values to the Workflow in order to help prevent breaking determinism.
```

Can't actually always prevent it

Reply:
It prevents this mechanism from interfering with the Workflow execution.